test/encryption.js

```diff
@@ -149,12 +152,329 @@ test('encryption - pass as promise', async (t) => {
+
+  await a.append('c can see me')
+
+  await remove(a, c, false)
+  await rotate(a, b4a.alloc(32, 1))
+
+  await a.append('c cannot see me')
+
+  await t.execution(replicateAndSync([a, b]))
+  await t.exception(replicateAndSync([a, c]))
+
+  t.alike(await a.view.get(0), 'c can see me')
+  t.alike(await b.view.get(0), 'c can see me')
+
+  t.alike(await a.view.get(1), 'c cannot see me')
+  t.alike(await b.view.get(1), 'c cannot see me')
+
+  await t.exception(c.view.get(1))
+
+  await a.close()
+  await b.close()
+})
+
+test('encryption - fast forward', async (t) => {
+  const encryptionKey = b4a.alloc(32).fill('secret')
+
+  const { bases } = await create(2, t, {
+    apply,
+    open,
+    encryptionKey,
+    fastForward: true
+  })
+
+  const [a, b] = bases
+
+  await add(a, b, false)
+
+  await replicateAndSync([a, b])
+  await b.append(null)
+  await replicateAndSync([a, b])
+
+  await rotate(a, b4a.alloc(32, 1))
+  for (let i = 0; i < 200; i++) {
+    await a.append('interval 1 - ' + i)
+  }
+
+  await rotate(a, b4a.alloc(32, 2))
+  for (let i = 0; i < 200; i++) {
+    await a.append('interval 2 - ' + i)
+  }
+
+  await rotate(a, b4a.alloc(32, 3))
+  for (let i = 0; i < 200; i++) {
+    await a.append('interval 3 - ' + i)
+  }
+
+  await rotate(a, b4a.alloc(32, 4))
+  for (let i = 0; i < 200; i++) {
+    await a.append('interval 4 - ' + i)
+  }
+  await rotate(a, b4a.alloc(32, 5))
+  for (let i = 0; i < 200; i++) {
+    await a.append('interval 5 - ' + i)
+  }
+
+  const ff = t.execution(
+    new Promise((resolve, reject) => {
+      b.on('fast-forward', pass)
+      const t = setTimeout(cleanup, 1000, false)
+
+      function pass() {
+        return cleanup(true)
+      }
+
+      function cleanup(ffed) {
+        b.removeListener('fast-foward', pass)
+        clearTimeout(t)
+
+        if (ffed) resolve()
+        else reject(new Error('timeout'))
+      }
+    })
+  )
+
+  await replicateAndSync([a, b])
+  await ff
+
+  t.is(a.view.length, 1000)
+  t.is(b.view.length, 1000)
+
+  t.alike(await a.view.get(999), 'interval 5 - 199')
+  t.alike(await b.view.get(999), 'interval 5 - 199')
+
+  await a.close()
+  await b.close()
+})
+
+test('encryption - rotate writer encryption', async (t) => {
+  const encryptionKey = b4a.alloc(32).fill('secret')
+
+  const { bases } = await create(2, t, {
+    apply,
+    open,
+    encryptionKey
+  })
+
+  const [a, b] = bases
+
+  await add(a, b, false)
+  await replicateAndSync([a, b])
+
+  await b.append(null)
+  await replicateAndSync([a, b])
+
+  await a.append('c can see me')
+
+  const newLocal = b.store.get({
+    manifest: {
+      version: 2,
+      signers: [
+        {
+          publicKey: b.local.keyPair.publicKey
+        }
+      ]
+    },
+    keyPair: b.local.keyPair
+  })
+
+  await newLocal.ready()
+
+  // rotate b writer
+  await remove(a, b, false)
+  await b.setLocal(newLocal.key)
+  await add(a, b, false)
+
+  await replicateAndSync([a, b])
+
+  await b.append('encrypted!')
```

Comment:
Do we want to assert that this block after the writer rotation is processed and in the view?